src/structures/command/parameters/types/UserParameter.js

```diff
@@ -13,31 +13,38 @@ module.exports = class UserParameter extends Parameter {
       acceptUser: defVal(options, 'acceptUser', true),
       acceptDeveloper: defVal(options, 'acceptDeveloper', true),
       acceptSelf: !!options.acceptSelf,
+      acceptPartial: !!options.acceptPartial,
       errors: {
         invalidUser: 'errors:invalidUser',
         acceptSelf: 'errors:sameUser',
         acceptBot: 'errors:invalidUserBot',
         acceptUser: 'errors:invalidUserNotBot',
         acceptDeveloper: 'errors:userCantBeDeveloper',
+        acceptPartial: 'errors:userPartial',
         ...(options.errors || {})
       }
     }
   }
 
-  static parse (arg, { t, client, author, guild }) {
+  static async parse (arg, { t, client, author, guild }) {
     if (!arg) return
 
     const regexResult = MENTION_REGEX.exec(arg)
     const id = regexResult && regexResult[1]
     const findMember = guild.members.cache.get(id) || guild.members.cache.find(m => m.user.username.toLowerCase().includes(arg.toLowerCase()) || m.displayName.toLowerCase().includes(arg.toLowerCase()))
+    try {
+      const partialUser = await client.users.fetch(id)
+      const user = client.users.cache.get(id) || (!!findMember && findMember.user)
+      if (!user && !this.acceptPartial) throw new CommandError(t(this.errors.invalidUser))
+      if (!this.acceptSelf && user.id === author.id) throw new CommandError(t(this.errors.acceptSelf))
+      if (!this.acceptBot && user.bot) throw new CommandError(t(this.errors.acceptBot))
+      if (!this.acceptUser && !user.bot) throw new CommandError(t(this.errors.acceptUser))
+      if (!this.acceptDeveloper && PermissionUtils.isDeveloper(client, user)) throw new CommandError(t(this.errors.acceptDeveloper), false)
+      if (!this.acceptPartial && partialUser && !findMember) throw new CommandError(t(this.errors.acceptPartial))
 
-    const user = client.users.cache.get(id) || (!!findMember && findMember.user)
-    if (!user) throw new CommandError(t(this.errors.invalidUser))
-    if (!this.acceptSelf && user.id === author.id) throw new CommandError(t(this.errors.acceptSelf))
-    if (!this.acceptBot && user.bot) throw new CommandError(t(this.errors.acceptBot))
-    if (!this.acceptUser && !user.bot) throw new CommandError(t(this.errors.acceptUser))
-    if (!this.acceptDeveloper && PermissionUtils.isDeveloper(client, user)) throw new CommandError(t(this.errors.acceptDeveloper), false)
-
-    return user
+      return this.acceptPartial ? partialUser : user

```

Comment:
```suggestion
      return user || partialUser

```

Reply:
This assumes that every command will always want a full user. This might limit what kind of command we are capable of creating in the future.

Reply:
Nope. Check my first requested change.